### DISCUSSION TITLE
Metalama in UWP project

### DISCUSSION THREAD

**Discussion by HakobyanAram** [metalama/Metalama#55] [Q&A] [answered]
Hello everyone! I want to use Metalama for the UWP project. I installed the `Metalama.Framework 0.5.66-preview` version from `NuGet`.
 The Target version in my project is 'Windows 10, version 2004 (10.0; Build 19041)` and the Min version is  'Windows 10, version 1809(10.0; Build 17763)`. After the compilation, I get the following error
 
![image](https://user-images.githubusercontent.com/51437851/204881384-2757f29b-f7b1-46f0-9eed-01383783f8fd.png)

Here is the code:
```
public class LogAttribute : OverrideMethodAspect
    {
       
        public override dynamic? OverrideMethod()
        {
            Debug.WriteLine($"Executing>>>>>>>>>>>>>>>>>>>>>>>>> {meta.Target.Method}.");

            return meta.Proceed();
        }
    }
```

```
   public sealed partial class MainPage : Page
    {
        public MainPage()
        {
            this.InitializeComponent();
            TestLog();
        
        }

        [Log]
        public int TestLog()
        {
            Debug.WriteLine("Test......................... " );
            return 5;
        }
  
    }
```

What do I  wrong?
Is there any sample UWP project?

**Comment by gfraiteur** [comment, marked as answer]
You have to manually set this in your csproj file:

```
<PropertyGroup>
  <LangVersion>10.0</LangVersion>
</PropertyGroup?
```

We don't have an example at the moment.

**Reply by gfraiteur** [reply]
Oops sorry I didn't look at the second error. It seems that the problem is with the CompileXaml task, not with your main project file.
I'll file a bug and follow up.

**Reply by gfraiteur** [reply]
I checked myself and:
* I could successfully use the INPC aspect in a UWP app
* The error you reported is indeed caused by incorrect language version. Setting LangVersion to 10.0 should work.

**Reply by HakobyanAram** [reply]
@gfraiteur Thanks for the quick response! I did it, and added the following In `<PropertyGroup>`
```
<LangVersion>10.0</LangVersion>
<Nullable>enable</Nullable>
```
I get an error but my UWP application is running. 

![image](https://user-images.githubusercontent.com/51437851/205130726-eb6187e1-1a46-448b-9909-82b7c287d083.png)

And you can see in the output window there is no text which writes `OverrideMethod`. There is only one text which corresponds `TestLog ` method.

Here are the error messages:
![image](https://user-images.githubusercontent.com/51437851/205131453-7303c1e6-63fc-49bb-a796-c0258e7836fc.png)

[Here is the project:](https://drive.google.com/file/d/1Crdq03aKLk_ivtVNumqQJfGBIvCXhtyP/view?usp=sharing)

**Reply by gfraiteur** [reply]
Thank you. I am working on that. I haven't been able to get success so far, even with the newly released build.

**Reply by gfraiteur** [reply]
This issue is now fixed.

**Reply by HakobyanAram** [reply]
Thanks a lot. The error message is disappeared. And LogAttribute.OverrideMethod works correctly. But two warnings still appear.

![image](https://user-images.githubusercontent.com/51437851/207137028-32f2d511-5c8e-493d-b640-ed7241571586.png)

**Reply by gfraiteur** [reply]
For the second warning, please create an issue and attach the files found in `%TEMP%\Metalama\CrashReports`.
